pkg/health/checker/checker.go

```diff
@@ -22,11 +23,16 @@ import (
 	klabels "k8s.io/kubernetes/pkg/labels"
 )
 
+// originally from pkg/ds but import cycle not allowed
+const (
+	DSIDLabel = "daemon_set_id"
```

Comment:
This is a sign that this package is poorly coupled to the daemon set package.

Could we integrate the node fetching in a way that's less tightly coupled? What if serviceNodesDS and serviceNodesRC was a function of type `function() ([]types.NodeName, error)` passed in by the client?

Reply:
I suppose serviceNodes can be a function passed to the constructor of the healthChecker.